ambry-clustermap/src/main/java/com/github/ambry/clustermap/ClusterMapUtils.java

```diff
@@ -943,6 +945,12 @@ private boolean areAllLocalReplicasForPartitionUp(PartitionId partitionId) {
           return false;
         }
       }
+
+      if(!(areAllReplicasForPartitionUp(partitionId) && areAllReplicaStatesEligibleForPut(partitionId, null)))
```

Comment:
We do not need to check if all replicas are available and eligible in the remote colos. We should just check if a subset available and eligible (maybe 1 replica in each remote colo being available and eligible is good enough).

Reply:
Yeah, you're right - I changed this.